tests/unit/plugins/dist_matrixops/dist_identity_2_loc.cpp

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2020 Nanmiao Wu
+// Copyright (c) 2020 Bita Hasheminezhad
+// Copyright (c) 2017-2020 Hartmut Kaiser
+
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/phylanx.hpp>
+
+#include <hpx/hpx_init.hpp>
+#include <hpx/include/lcos.hpp>
+#include <hpx/testing.hpp>
+
+#include <array>
+#include <cstdint>
+#include <string>
+
+#include <utility>
+#include <vector>
+
+#include <blaze/Math.h>
+#include <blaze_tensor/Math.h>
+
+///////////////////////////////////////////////////////////////////////////////
+phylanx::execution_tree::primitive_argument_type compile_and_run(
+    std::string const& name, std::string const& codestr)
+{
+    phylanx::execution_tree::compiler::function_list snippets;
+    phylanx::execution_tree::compiler::environment env =
+        phylanx::execution_tree::compiler::default_environment();
+
+    auto const& code =
+        phylanx::execution_tree::compile(name, codestr, snippets, env);
+    return code.run().arg_;
+}
+
+void test_identity_d_operation(std::string const& name, std::string const& code,
+    std::string const& expected_str)
+{
+    phylanx::execution_tree::primitive_argument_type result =
+        compile_and_run(name, code);
+    phylanx::execution_tree::primitive_argument_type comparison =
+        compile_and_run(name, expected_str);
+
+    // comparing annotations
+    HPX_TEST_EQ(*(result.annotation()), *(comparison.annotation()));
```

Comment:
We need to compare results and comparison, since we need their values to be the same too. Please change this to `HPX_TEST_EQ(result, comparison);`

Reply:
Changed to HPX_TEST_EQ(result, comparison).